### PR TITLE
adding smoke tests for v23

### PR DESCRIPTION
**Pull request by shileiwill**
AUTO-9980

This PR adds integration test for Automation v23, using the LINK token and using Conditional upkeeps. Will add more use cases in following PRs, e.g. other ERC20 tokens and log trigger based upkeeps

 > SETH_LOG_LEVEL=trace make test_smoke_product product="automation" args="-timeout 20m -test.run TestAutomationBasic/registry_2_3_condi"

The integration test passes.

### CODE REVIEW

File: integration-tests/go.mod [hunk 1]
```diff
@@ -22,19 +22,19 @@ require (
...
-	github.com/rs/zerolog v1.31.0
+	github.com/rs/zerolog v1.30.0
 	github.com/scylladb/go-reflectx v1.0.1
 	github.com/segmentio/ksuid v1.0.4
 	github.com/shopspring/decimal v1.3.1
 	github.com/slack-go/slack v0.12.2
 	github.com/smartcontractkit/chainlink-automation v1.0.4
 	github.com/smartcontractkit/chainlink-common v0.1.7-0.20240620164046-1c03d0a97b20
-	github.com/smartcontractkit/chainlink-testing-framework v1.31.1
+	github.com/smartcontractkit/chainlink-testing-framework v1.30.5
```
Comment (anirudhwarrier) [inline integration-tests/go.mod]: @shileiwill can you undo this CTF downgrade change

Comment (anirudhwarrier) [reply integration-tests/go.mod]: This also could be linked - https://github.com/smartcontractkit/chainlink/actions/runs/9609686757/job/26504664318?pr=13644#step:4:770

File: integration-tests/contracts/ethereum_keeper_contracts.go [hunk 2]
```diff
@@ -150,6 +153,27 @@ type KeeperRegistrySettings struct {
...
+		MaxPerformGas:          rcs.MaxPerformGas,
+		MaxCheckDataSize:       rcs.MaxCheckDataSize,
+		MaxPerformDataSize:     rcs.MaxPerformDataSize,
+		MaxRevertDataSize:      rcs.MaxRevertDataSize,
+		FallbackGasPrice:       rcs.FallbackGasPrice,
+		FallbackLinkPrice:      rcs.FallbackLinkPrice,
+		Transcoder:             common.Address{},
+		Registrars:             []common.Address{common.HexToAddress(registrar)},
+		UpkeepPrivilegeManager: registryOwnerAddress,
+		ChainModule:            chainModuleAddress, // TODO here is the chainModule
```
Comment (anirudhwarrier) [inline integration-tests/contracts/ethereum_keeper_contracts.go]: nit: what is this `TODO`?

**Review by anirudhwarrier — CHANGES_REQUESTED** [review]
left some comments

File: integration-tests/contracts/ethereum_keeper_contracts.go (hunk 2 above)
Comment (shileiwill) [reply integration-tests/contracts/ethereum_keeper_contracts.go]: should be removed, good catch.

**Review by anirudhwarrier — APPROVED** [review]

**Review by FelixFan1992 — APPROVED** [review]

**Pull request merged**